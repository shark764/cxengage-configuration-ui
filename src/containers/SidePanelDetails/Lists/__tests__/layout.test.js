/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import ListsDetailsPanel from '../layout';

describe('<ListsDetailsPanel />', () => {
  let customTableItems;
  let customTableFields;
  beforeEach(() => {
    customTableItems = [
      {
        tableField1: 'test 1',
        tableField2: 'test 2',
        tableField3: 'test 3',
        tableField4: 'test 4'
      }
    ];
    customTableFields = [
      {
        type: 'string',
        name: 'tableField1',
        label: 'label 1',
        required: true
      },
      {
        type: 'string',
        name: 'tableField2',
        label: 'label 2',
        required: true
      },
      {
        type: 'string',
        name: 'tableField3',
        label: 'label 3',
        required: true
      },
      {
        type: 'string',
        name: 'tableField4',
        label: 'label 4',
        required: true
      }
    ];
  });
  it('renders lists detailsPanel', () => {
    const rendered = shallow(
      <ListsDetailsPanel
        id="7a96c534-31f9-11n8-88b9-9440dab83f21"
        className="details-panel"
        userHasUpdatePermission={true}
        alertMessage=""
        children={'Mock Child'}
        listType="Reason Codes Type"
        tableItems={customTableItems}
        tableFields={customTableFields}
        openCreateListItemModal={() => {}}
        updateSubEntity={() => {}}
        deleteSubEntity={() => {}}
        inherited={true}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
