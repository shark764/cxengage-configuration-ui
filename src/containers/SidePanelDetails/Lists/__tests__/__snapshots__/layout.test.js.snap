// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListsDetailsPanel /> renders lists detailsPanel 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<ListsDetailsPanel /> renders lists detailsPanel not inherited 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<ListsDetailsPanel /> renders lists detailsPanel with alert message 1`] = `
ShallowWrapper {
  "length": 1,
}
`;
